### PR TITLE
Print warnings on CREATE or UPGRADE

### PR DESCRIPTION
**Pull request by sfc-gh-cgorrie**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
When warnings are returned by CREATE APPLICATION or UPGRADE APPLICATION, show them using `cc.warning` as well as an extra line break to draw visual attention to the warnings.

image

### CODE REVIEW

File: tests/nativeapp/test_run_processor.py [hunk 1]
```diff
@@ -202,6 +203,72 @@ def test_create_dev_app_create_new_w_no_additional_privileges(
     assert mock_execute.mock_calls == expected
 
 
+# Test create_dev_app with no existing application AND create returns a warning
+@mock.patch(RUN_PROCESSOR_GET_EXISTING_APP_INFO, return_value=None)
+@mock.patch(NATIVEAPP_MANAGER_EXECUTE)
+@mock.patch(f"{NATIVEAPP_MODULE}.cc.warning")
+@mock_connection()
+def test_create_dev_app_with_warning(
```
Comment (sfc-gh-pjafari) [inline tests/nativeapp/test_run_processor.py:211]: should we add a test for the upgrade case as well?

File: src/snowflake/cli/_plugins/nativeapp/run_processor.py [hunk 2]
```diff
@@ -72,6 +72,22 @@
...
+def print_warnings(create_or_upgrade_cursor: Optional[SnowflakeCursor]):
+    """
+    Shows warnings in the console returned by the CREATE or UPGRADE
+    APPLICATION command.
+    """
+    if not create_or_upgrade_cursor:
+        return
+
+    messages = [row[0] for row in create_or_upgrade_cursor.fetchall()]
+    warnings = [m for m in messages if m.lower().startswith("warning:")]
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/nativeapp/run_processor.py:84]: Is it guaranteed that all warnings will start with this string?
Maybe you can add a sanity integration test?

File: src/snowflake/cli/_plugins/nativeapp/run_processor.py [hunk 3]
```diff
@@ -72,6 +72,22 @@
...
+    if warnings:
+        for warning in warnings:
+            cc.warning(warning)
+        cc.message("")
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/nativeapp/run_processor.py:88]: Why is this needed?

Comment (sfc-gh-cgorrie) [reply src/snowflake/cli/_plugins/nativeapp/run_processor.py:88]: As mentioned in the PR description, it's to make it easy to tell there are warnings. There's so much irrelevant output it's hard to actually see this, which is very high signal, vs a lot of the rest of the noise.

File: src/snowflake/cli/_plugins/nativeapp/run_processor.py (hunk 2 above)
Comment (sfc-gh-cgorrie) [reply src/snowflake/cli/_plugins/nativeapp/run_processor.py:84]: I don't think this requires an integration test; I don't want it to start failing if slightly different warnings / more warnings are added.

Comment (sfc-gh-cgorrie) [reply src/snowflake/cli/_plugins/nativeapp/run_processor.py:84]: Checking in on the `Warning:` thing now, maybe I can just show the "created successfully" message too (was trying to cut down on the amount of lines printed)

File: src/snowflake/cli/_plugins/nativeapp/run_processor.py [hunk 4]
```diff
@@ -72,20 +72,18 @@
...
         return
 
     messages = [row[0] for row in create_or_upgrade_cursor.fetchall()]
-    warnings = [m for m in messages if m.lower().startswith("warning:")]
-    if warnings:
-        for warning in warnings:
-            cc.warning(warning)
-        cc.message("")
+    for message in messages:
+        cc.warning(message)
```
Comment (sfc-gh-pjafari) [inline src/snowflake/cli/_plugins/nativeapp/run_processor.py:85]: is `warning` intended here for all messages?

**Review by sfc-gh-fcampbell — APPROVED** [review]

File: src/snowflake/cli/_plugins/nativeapp/run_processor.py (hunk 4 above)
Comment (sfc-gh-cgorrie) [reply src/snowflake/cli/_plugins/nativeapp/run_processor.py:85]: For posterity: rather than try to determine what's a warning or not, we'll tolerate false positives (e.g. the `Application successfully created` message).
